lib/compute/jenkins-main-node.ts

```diff
@@ -329,6 +349,19 @@ export class JenkinsMainNode {
       // Therefore, sleep 60 seconds to wait for plugins to install and jenkins to start which is required for the next step
       InitCommand.shellCommand('sleep 60'),
 
+      // Create groovy script that holds the agent Node config for EC2 plugin ref:https://gist.github.com/vrivellino/97954495938e38421ba4504049fd44ea
+      agentNode.asInitFile('/agent-node-al2-x64.groovy', agentNodeProps, al2x64AgentNodeConfig),
```

Comment:
Can we pass in an `AgentNode` object with a method that produces these InitElement and adds them to the end of the list?  It would reduce the amount of argument passing by a wide margin and reduce coupling

line 137:
```
CloudFormationInit.fromElements(...JenkinsMainNode.configElements(...),
                                ...props.cloudAgent.configElements())
```

Reply:
Done